zkml/src/layers/provable/node.rs

```diff
@@ -0,0 +1,401 @@
+use anyhow::{Result, anyhow, bail, ensure};
```

Comment:
This is like a temporary file? Most of the code seems duplicated with the one in `mod.rs`